pkg/cmd/pr/merge/merge.go

```diff
@@ -175,6 +177,7 @@ func NewCmdMerge(f *cmdutil.Factory, runF func(*MergeOptions) error) *cobra.Comm
 	cmd.Flags().BoolVarP(&flagSquash, "squash", "s", false, "Squash the commits into one commit and merge it into the base branch")
 	cmd.Flags().BoolVar(&opts.AutoMergeEnable, "auto", false, "Automatically merge only after necessary requirements are met")
 	cmd.Flags().BoolVar(&opts.AutoMergeDisable, "disable-auto", false, "Disable auto-merge for this pull request")
+	cmd.Flags().StringVar(&opts.Sha, "sha", "", "`SHA` that pull request head must match to allow merge")
```

Comment:
I don't know if `--sha` is descriptive enough of a name here in the context of a pull request, I feel like we could make it more clear what this flag is going to do. What do you think about `--match-head-commit`? It is a bit more verbose but I think it conveys the purpose of the flag.

Reply:
Thanks @samcoe for the help and feedback. I've done the following:

- updated `sha` to `match-head commit` (I agree I like that a lot better)
- removed the interactive portion of the code
- added a test to ensure `expectedHeadOid` is passed when a value is passed to `--match-head-commit`

Question:

Would it be appropriate to add `assert.NotContains(t, input, "expectedHeadOid")` to the other tests similar to how `assert.NotContains(t, input, "commitHeadline")` is in every test?